compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,32 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        def warnIfDuplicatedKey(keys: List[String], namesSeen: Set[String] = Set.empty): Unit = keys match {
+          case key :: tail =>
+            if (namesSeen(key))
+              reporter.warning(pos, s"Duplicate key in object literal: $key. Only the last occurrence is assigned.")
+            warnIfDuplicatedKey(tail, namesSeen + key)
+          case Nil => // Do nothing and finish
+        }
+
         // Extract first arg to future proof against varargs
         extractFirstArg(genArgs) match {
           // case js.Dynamic.literal("name1" -> ..., "name2" -> ...)
           case (js.StringLiteral("apply"), jse.LitNamed(pairs)) =>
+            warnIfDuplicatedKey(pairs map (_._1.name))
```

Comment:
We have a history of remaining `xs map (...)`, but the new style, if you use parentheses, is `pairs.map(_._1.name)`.